### PR TITLE
Add package comments to packages that are missing them

### PR DESCRIPTION
**Pull request by wking**
I've been chipping away at this over the past two days while I survey
the repository.  Currently stumping me for a description is
core/corerouting, so I'd welcome some ideas for a brief description
there.  Otherwise, I'm trying to describe these packages based on a
quick skim, so feedback/clarification/fixes/etc. are welcome :).

I still have a few more packages to document, but I thought I'd push
this PR now to start getting feedback on the earlier docs while I work
through the remaining packages.

### CODE REVIEW

**Comment by whyrusleeping** [comment]
:clap: fighting the good fight!

File: doc.go [hunk 1]
```diff
@@ -1,2 +1,27 @@
...
+There are sub-packages within the ipfs package for various low-level
+utilities, which are in turn assembled into:
+
+core/...:
+  The low-level API that gives consumers all the knobs they need,
+  which we try hard to keep stable.
+shell/...:
+  The high-level API that gives consumers easy access to common
+  operations (e.g. create a file node from a reader without wrapping
+  with metadata). We work really hard to keep this stable.
```
Comment (jbenet) [inline doc.go]: :+1: :clap: 

Maybe also include:

```go
routing/...:
  Routing system interface and implementations.
  For example, the IPFS DHT lives here.

exchange/...:
  Exchange system interface and implementations
  For example, Bitswap lives here.
```

**Comment by jbenet** [comment]
:+1: LGTM

**Comment by wking** [comment]
On Sat, May 02, 2015 at 12:22:18PM -0700, Juan Batiz-Benet wrote:
> Merged #1185.

I guess I'll submit the next round of godocs as a follow-up PR ;).

**Comment by jbenet** [comment]
Yeah, keeps the ball rolling. always nice to merge things :) :+1:

**Pull request merged**